src/otk/transform.py

```diff
@@ -53,6 +54,42 @@ def construct_mapping(self, node, deep=False):
         return super().construct_mapping(node, deep)
 
 
+def _find_data(node_list, key):
+    """ Helper function to find the yaml ScalarNode containing
+    our context information for HiddenAttrList and HiddenAttrDict
+    """
+    for node in node_list:
+        if node[0].value == key:
+            return node[0]
+    raise KeyError(key)
+
+
+def _add_hidden_attributes(obj, key, key_data):
+    line_number = key_data.start_mark.line + 1
+    column = key_data.start_mark.column + 1
+    filename = os.path.relpath(key_data.start_mark.name, os.path.curdir)
+    obj.set_attribute(key, "src", f"{filename}:{line_number}")
+    obj.set_attribute(key, "filename", filename)
+    obj.set_attribute(key, "linenumber", line_number)
+    obj.set_attribute(key, "column", column)
+
+
+def hidden_attr_dict_constructor(loader, node):
+    data = loader.construct_mapping(node, deep=True)
+    hidden_data_node = HiddenAttrDict(data)
+    for key, _ in hidden_data_node.items():
+        key_data = _find_data(node.value, key)
+        _add_hidden_attributes(hidden_data_node, key, key_data)
+    return hidden_data_node
+
+
+def hidden_attr_list_constructor(loader, node):
+    data = loader.construct_sequence(node, deep=True)
+    hidden_data_node = HiddenAttrList(data)
+    _add_hidden_attributes(hidden_data_node, "self", node)
```

Comment:
For example, this could be `HiddenAttrList.with_hidden_attributes(data, hidden_data_node, "self", node)` or such if the above as a classmethod.

Reply:
The idea was that the `HiddenAttr*` objects are generic and this is a usecase specific type/usage
That's why it's in `transform.py` but for sure `transform.py` looks nicer when this is a classmethod